addons/stock_account/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,256 @@
+        ORDER BY pph.datetime, pph.id
+    """, (company_id, product_id))
+    return env.cr.dictfetchall()
+
+
+def get_stock_moves(env, company_id, product_id):
+    env.cr.execute("""
+        WITH account_move_rel AS (
+            SELECT id, stock_move_id
+            FROM (
+                SELECT id, stock_move_id, COUNT(*) OVER(PARTITION BY stock_move_id) AS qty
+                FROM account_move
+                WHERE stock_move_id IS NOT NULL
+            ) foo
+            WHERE qty = 1
+        )
+        SELECT sm.id, sm.company_id, sm.product_id, sm.date, sm.product_qty, sm.reference,
+            COALESCE(sm.price_unit, 0.0) AS price_unit, rel.id AS account_move_id,
+            sm.create_uid, sm.create_date, sm.write_uid, sm.write_date,
+            CASE WHEN (sl.usage <> 'internal' AND (sl.usage <> 'transit' OR sl.company_id <> sm.company_id))
+                   AND (sld.usage = 'internal' OR (sld.usage = 'transit' AND sld.company_id = sm.company_id))
+                   THEN 'in'
+                 WHEN (sl.usage = 'internal' OR (sl.usage = 'transit' AND sl.company_id = sm.company_id))
+                   AND (sld.usage <> 'internal' AND (sld.usage <> 'transit' OR sld.company_id <> sm.company_id))
+                   THEN 'out'
+                 WHEN sl.usage = 'supplier' AND sld.usage = 'customer' THEN 'dropship'
+                 WHEN sl.usage = 'customer' AND sld.usage = 'supplier' THEN 'dropship_return'
+                 ELSE 'other'
+            END AS move_type
+        FROM stock_move sm
+        LEFT JOIN stock_location sl ON sl.id = sm.location_id
+        LEFT JOIN stock_location sld ON sld.id = sm.location_dest_id
+        LEFT JOIN account_move_rel rel ON rel.stock_move_id = sm.id
+        WHERE sm.company_id = %s AND sm.product_id = %s AND state = 'done'
+        ORDER BY sm.date, sm.id
+    """, (company_id, product_id))
+    return env.cr.dictfetchall()
+
+
+@openupgrade.logging()
+def generate_stock_valuation_layer(env):
+    openupgrade.logged_query(
+        env.cr, """
+            ALTER TABLE stock_valuation_layer
+            ADD COLUMN old_product_price_history_id integer""",
+    )
+    company_obj = env["res.company"]
+    product_obj = env["product.product"]
+    # Needed to modify global variable
+    global precision_price
+    precision_price = env["decimal.precision"].precision_get("Product Price")
+    precision_uom = env["decimal.precision"].precision_get(
+        "Product Unit of Measure"
+    )
+    companies = company_obj.search([])
+    products = product_obj.with_context(active_test=False).search([("type", "in", ("product", "consu"))])
+    all_svl_list = []
+    for product in products:
+        for company in companies:
+            history_lines = get_product_price_history(env, company.id, product.id)
+            moves = get_stock_moves(env, company.id, product.id)
+            svl_in_vals_list = []
+            svl_out_vals_list = []
+            svl_man_vals_list = []
```

Comment:
Can you explain what does this list of layer contain? for the svl_in_vals_list and svl_out_vals_list is clear the purpose but this one is unclear for me. Thanks in advance!